### PR TITLE
Cleanup, Refactoring, and Code Comments In Preparation for First Pull Request to Main Weaviate Repo

### PR DESCRIPTION
**Pull request by sourcesync**
### What's being changed:

* Added more code comments
* Added active links from various README's
* Improved language in READMEs
* Ensured the toplevel Dockerfile can be used to build the main Weaviate container

### Review checklist

* Start with the main [gsi/README](./gsi/README.md) and make sure the links work correctly
* Check for spelling and awkward langage
* Test the instructions in the "Quick Start" section

### CODE REVIEW

File: gsi/tests/weaviate-gemini-sanity.py
Comment (iamsabhoho) [inline gsi/tests/weaviate-gemini-sanity.py:1]: `  File "weaviate-gemini-sanity.py", line 1
    //                           _       _
    ^
SyntaxError: invalid syntax`

syntax error in gsi/tests/weaviate-gemini-sanity.py

**Review by iamsabhoho — CHANGES_REQUESTED** [review]
syntax error in gsi/tests/weaviate-gemini-sanity.py

File: gsi/tests/weaviate-gemini-sanity.py
Comment (sourcesync) [reply gsi/tests/weaviate-gemini-sanity.py:1]: Thanks @iamsabhoho.  I pushed a fix.

File: gsi/README.md
Comment (iamsabhoho) [inline gsi/README.md:1]: "web service"

File: gsi/README.md [hunk 1]
```diff
@@ -8,14 +8,14 @@ The features of this plugin include:
 * the plugin is implemented in pure Golang
 * the plugin interfaces directly with Gemini's FVS (Fast Vector Search) REST webservice API
```
Comment (iamsabhoho) [inline gsi/README.md:9]: "web service"

File: gsi/README.md [hunk 2]
```diff
@@ -29,23 +29,23 @@ If you just want to see something working first, then following the directions i
...
-* Run the docker build script "rundockerbuild.sh" which builds a vectorization container locally called "gsi/t2v"
-* Now cd into the "gsi/docker" directory
-* Edit the file "docker-compose-sanity.yml" with your allocation id and run it via docker-compose.
+* Now cd into the [gsi](../gsi) directory
+* Run the docker build script [rundockerbuild.sh](./rundockerbuild.sh) which builds a Gemini Plugin-enabled container locally called "gsi/weaviate"
+* Now cd into the [gsi/docker/t2v](docker/t2v) directory
+* Run the docker build script [rundockerbuild.sh](docker/t2v/rundockerbuild.sh) which builds a vectorization container locally called "gsi/t2v"
+* Now cd into the [gsi/docker](./docker) directory
+* Edit the file [docker-compose-sanity.yml](./docker/docker-compose-sanity.yml) and set the env var GEMINI_ALLOCATION_ID to your allocation id and run the file via docker-compose.
 * Wait until you see the startup message which indicates that the Weaviate server has successfully start.
```
Comment (iamsabhoho) [inline gsi/README.md:43]: started

File: gsi/README.md [hunk 3]
```diff
@@ -58,72 +58,47 @@ This codebase contains all the Gemini Plugin code, so build and deploy the conta
 
 The following environmental variables are unique to Gemini Plugin configuration at the server:
 * *GEMINI_ALLOCATION_ID*  -This must be set to a valid allocation id.  Please consult your onboarding instructions or reach out to your GSI support contact for more information.
-* *GEMINI_DATA_DIRECTORY* -This directory must exist and live under "/home/public" (it's an FVS requirement.)  In addition to the FVS API, this directory is critical for data transfer from the Gemini Plugin.
+* *GEMINI_DATA_DIRECTORY* -This directory must exist and live under "/home/public."  This location is currently an FVS requirement.  This directory itself is critical for data transfer betweeen Gemini Plugin and the FVS.
```
Comment (iamsabhoho) [inline gsi/README.md:61]: between, there is an extra e

File: gsi/README.md [hunk 4]
```diff
@@ -58,72 +58,47 @@ This codebase contains all the Gemini Plugin code, so build and deploy the conta
...
 * *GEMINI_FVS_SERVER* -This is the server address of your FVS server.  Typically, its value should be 'localhost' since the FVS server would typically be co-located with your Weaviate server instance.
 * *GEMINI_DEBUG* -By default its false.  Set to 'true' (lower-case) to see more Gemini Plugin log messages printed to the console.  This is useful when debugging Gemini Plugin configuration issues.
```
Comment (iamsabhoho) [inline gsi/README.md:63]: it's

File: gsi/README.md [hunk 5]
```diff
@@ -58,72 +58,47 @@ This codebase contains all the Gemini Plugin code, so build and deploy the conta
...
+There is a python example located at [gsi/tests/weaviate-gemini-sanity.py](./tests/weaviate-gemini-sanity.py) that demonstrates this.
 
-# Benchmarks
+# Comparing Gemini Plugin To Weaviate's Native HNSW
 
-## Gemini Fast Vector Search (FVS)
+## Search Performance
 
-The Weaviate Gemini Plugin communicates with GSI's Fast Vector Search (FVS) REST web service.
+Since Weaviate can now support both the Gemini Plugin alongside it's native HNSW, it's now straightforward to perform an apples-to-apples comparison of both ANN algorithms on the same system and the same data.  We will be performing these measurements on various large scale datasets and reporting the results here at a later time.
```
Comment (iamsabhoho) [inline gsi/README.md:77]: its

Comment (iamsabhoho) [inline gsi/README.md:77]: alongside its?

File: gsi/README.md [hunk 6]
```diff
@@ -58,72 +58,47 @@ This codebase contains all the Gemini Plugin code, so build and deploy the conta
...
+As datasets become larger, the elapsed time that your application needs to wait for the index training to complete will increase as well.  Please see the [Gemini offline training benchmarks](./fvs/README.md) for more information.
 
-This section will eventually contain resolutions to common problems that you may run into when either configuring the Gemini Plugin, or working with a Gemini index, the FVS system, or the APU hardware.
+#  Roadmap
 
+More Gemini Plugin improvements are coming as we continue to integrate deeper into Weaviate and, importantly, as the Weaviate ecosystem starts to give us feedback:
 
+* we are working on eliminating the off-line training process of the Gemini index
+* integrate system observability metrics into Prometheus alongside Weaviate
+* support Weaviate's fail-over and replication strategies for datacenter high-availablity deployment
```
Comment (iamsabhoho) [inline gsi/README.md:101]: data center

**Review by iamsabhoho — COMMENTED** [review]
- [x] Start with the main [gsi/README](https://github.com/sourcesync/weaviate/pull/gsi/README.md) and make sure the links work correctly

- [x] Check for spelling and awkward langage

- [x] Test the instructions in the "Quick Start" section

File: gsi/README.md (hunk 1 above)
Comment (sourcesync) [reply gsi/README.md]: fixed and pushed an update

File: gsi/README.md (hunk 2 above)
Comment (sourcesync) [reply gsi/README.md]: fixed and pushed

File: gsi/README.md (hunk 3 above)
Comment (sourcesync) [reply gsi/README.md]: fixed and pushed

File: gsi/README.md (hunk 6 above)
Comment (sourcesync) [reply gsi/README.md]: fixed and pushed

File: gsi/README.md [hunk 7]
```diff
@@ -58,72 +58,47 @@ This codebase contains all the Gemini Plugin code, so build and deploy the conta
...
+* implement Weaviate's native HNSW data recovery such as index re-initialization
+* integrate test automation including unit tests, stress tests, code coverage tests by extending Github's CI/CD infrastructure to GSI's hardware cloud
```
[the remark this answers is not in the document]
Comment (sourcesync) [reply gsi/README.md]: fixed and pushed

File: gsi/README.md (hunk 5 above)
Comment (sourcesync) [reply gsi/README.md]: fixed and pushed

**Comment by iamsabhoho** [comment]
under README.md, `Additional component descriptions:`

should **numpy** be **Numpy**?

**Review by iamsabhoho — APPROVED** [review]
lgtm!

**Pull request merged**